Content.Server/Bed/Sleep/SleepingSystem.cs

```diff
@@ -67,7 +67,10 @@ private void OnSleepStateChanged(EntityUid uid, MobStateComponent component, Sle
                 if (TryComp<SleepEmitSoundComponent>(uid, out var sleepSound))
                 {
                     var emitSound = EnsureComp<SpamEmitSoundComponent>(uid);
-                    emitSound.Sound = sleepSound.Snore;
+                    if (HasComp<SnoringComponent>(uid))
+                    {
+                        emitSound.Sound = sleepSound.Snore;
+                    }
```

Comment:
This will still cause the "Zzzz" popup message to display. Not sure if that's desired. On the one hand, it indicates snoring, so it shouldn't be there if you're not snoring. On the other hand, it's a handy indicator that someone is asleep.

Reply:
I'm more inclined to the second option